### PR TITLE
Update DevTools icons

### PR DESCRIPTION
**Pull request by birtles**
This reflects the changes from [bug 1474224](https://bugzilla.mozilla.org/show_bug.cgi?id=1474224).

In particular:

* Makes the "dock to bottom" icon have the same width as the other docking icons
* Updates the "undock" icon so that it matches the Photon "Open in New" icon but has the same proportions as the other docking icons
* Updates the "options" icon to follow the Photon "Settings" icon and removes the deprecated label

These icons don't include an explicit width/height since that's generally an anti-pattern when creating SVG resources (they should scale to their container) but perhaps it's needed for the Photon icon viewer
though?

### CODE REVIEW

File: photon-icons.json [hunk 1]
```diff
@@ -1923,7 +1923,7 @@
       }
     },
     { "name": "Tool Options",
```
Comment (bwinton) [inline photon-icons.json]: I know the svg file has a different name, but is this the same icon as https://design.firefox.com/icons/viewer/#preferences%20mac (`icons/desktop/preferences-16-macos.svg`)? And if so, can we use that file here, too?

**Comment by bwinton** [comment]
And yes, the icons will need explicit width/heights, or we get validation errors (which you can see at `npm run test`).

**Comment by aminalhazwani** [comment]
> And yes, the icons will need explicit width/heights, or we get validation errors (which you can see at npm run test).

And also because the icons are meant to be only used at their specific size even if you might scale the SVGs. The main reasons: icon-size consistency - all the icons are 16x16. And pixel snapping - being drawn at 16x16 we made sure that the stroke is snapped to the pixel grid. If we would scale the icon to 20x20 for example the two pixel stroke would be some like 2,xxx.

File: icons/desktop/dock-bottom-16.svg [hunk 2]
```diff
@@ -1,4 +1,10 @@
 <!-- This Source Code Form is subject to the terms of the Mozilla Public
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
-<svg viewBox="0 0 16 16" width="16" height="16" xmlns="http://www.w3.org/2000/svg" fill="context-fill #0b0b0b"><path d="M12 0a3 3 0 0 1 3 3v10a3 3 0 0 1-3 3H4a3 3 0 0 1-3-3V3a3 3 0 0 1 3-3h8zm0 2H4a1 1 0 0 0-1 1v7h10V3a1 1 0 0 0-1-1zm1 9H3v2a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1z"/></svg>
\ No newline at end of file
+<svg viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"
+  fill="context-fill #0b0b0b">
```
Comment (aminalhazwani) [inline icons/desktop/dock-bottom-16.svg]: Fill property should be only `fill="context-fill"`

File: icons/desktop/dock-undock-16.svg [hunk 3]
```diff
@@ -1,4 +1,11 @@
 <!-- This Source Code Form is subject to the terms of the Mozilla Public
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
-<svg viewBox="0 0 16 16" width="16" height="16" xmlns="http://www.w3.org/2000/svg" fill="context-fill #0b0b0b"><path d="M13 0a3 3 0 0 1 3 3v5a3 3 0 0 1-3 3H8a3 3 0 0 1-3-3V3a3 3 0 0 1 3-3zm0 2H8a1 1 0 0 0-1 1v5a1 1 0 0 0 1 1h5a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1z"/><path d="M9 13v-1h2v1a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V8a3 3 0 0 1 3-3h1v2H3a1 1 0 0 0-1 1v5a1 1 0 0 0 1 1h5a1 1 0 0 0 1-1z"/></svg>
\ No newline at end of file
+<svg viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"
+  fill="context-fill #0b0b0b">
```
Comment (aminalhazwani) [inline icons/desktop/dock-undock-16.svg]: Fill property should be only `fill="context-fill"`

File: icons/desktop/tool-options-16.svg [hunk 4]
```diff
@@ -1,4 +1,17 @@
...
\ No newline at end of file
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"
+  fill="context-fill #0b0b0b">
```
Comment (aminalhazwani) [inline icons/desktop/tool-options-16.svg]: Fill property should be only `fill="context-fill"`

File: photon-icons.json (hunk 1 above)
Comment (aminalhazwani) [reply photon-icons.json]: Also for the dock-undock we could use this icon https://github.com/FirefoxUX/photon-icons/blob/master/icons/desktop/open-in-new-16.svg

**Comment by birtles** [comment]
> And yes, the icons will need explicit width/heights, or we get validation errors (which you can see at npm run test).

Great, I'll fix that. I didn't know about `npm run test`.

File: icons/desktop/dock-bottom-16.svg (hunk 2 above)
Comment (birtles) [reply icons/desktop/dock-bottom-16.svg]: Ok. Is that so the users don't mistakenly assume they're getting the `context-fill` when they're actually getting the fallback?

**Comment by birtles** [comment]
> Also for the dock-undock we could use this icon https://github.com/FirefoxUX/photon-icons/blob/master/icons/desktop/open-in-new-16.svg

I'm afraid we need to match the proportions of the other dock icons (as per the commit message).

File: photon-icons.json (hunk 1 above)
Comment (birtles) [reply photon-icons.json]: Yep, let me compare those files tomorrow and merge/rename them as appropriate.

Unfortunately for the dock-undock one we need to match the proportions of the other dock icons so we can't use "Open in New" as-is.

**Comment by birtles** [comment]
I'll make the changes tomorrow.

**Comment by birtles** [comment]
I've addressed all the review feedback and added an extra commit that renames "Preferences Macos" to just "Preferences".

**Comment by aminalhazwani** [comment]
@birtles thanks for all the updates 🙌  yesterday we forgot to address this concern of yours:

> Updates the "undock" icon so that it matches the Photon "Open in New" icon but has the same proportions as the other docking icons

We tested both variants and we don't think it's necessary to create a slightly different icon to match the others icons ratio. If that would be true we would need to adjust many icons in many different context which is not ideal. Having icons in different ratios helps users quicker scan the content of a component. 

screen shot 2018-07-27 at 12 49 08 pm

**Comment by birtles** [comment]
Ok. It looks a bit clumsy to me, but I think once we arrange these icons in a row in [bug 1474223](https://bugzilla.mozilla.org/show_bug.cgi?id=1474223) it will be fine. So, let's not add this icon to the repo for now.

**Comment by birtles** [comment]
Ok, I've updated the PR. Please take a look.

**Pull request merged**